akka-http/src/main/scala/akka/http/scaladsl/common/NameReceptacle.scala

```diff
@@ -23,7 +23,7 @@ class NameReceptacle[T](val name: String) {
 }
 
 class NameUnmarshallerReceptacle[T](val name: String, val um: FSU[T]) {
-  def as[B](unmarshaller: Unmarshaller[T, B]) =
+  def as[B](implicit unmarshaller: Unmarshaller[T, B]) =
```

Comment:
Thanks!
I wish I could write `def as[B](implicit unmarshaller: Unmarshaller[T, B])(implicit fsu: FSU[T])` for `NameReceptacle` too :-) 
I need a time to think. But It looks like `FSU` is inevitable because the whole unmarshalling chain always starts from `NameReceptacle` and the composed transformer that I construct with `transform[B]` in `as[B]` always starts with `String` as an input type, in order to make the `ParamDef` implicits work correctly (https://github.com/akka/akka/blob/master/akka-http/src/main/scala/akka/http/scaladsl/server/directives/ParameterDirectives.scala#L124).